src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -590,7 +613,7 @@ GenerateAllAttributeEquivalences(PlannerRestrictionContext *plannerRestrictionCo
 		plannerRestrictionContext->joinRestrictionContext;
 
 	/* reset the equivalence id counter per call to prevent overflows */
```

Comment:
This is a bit of an anti-pattern, especially in the recursive world of query planning. We should consider passing down a pointer (e.g. to a struct containing the counter) or adding it to the restriction context structs or assigning the IDs later.